src/App.tsx

```diff
@@ -1,56 +1,88 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { debounce } from 'lodash';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './DropdownMenu';
+import { DropdownInput } from './DropdownInput';
+
+const search = (people: Person[], query: string) => {
+  const preparedQuery = query.trim().toLowerCase();
+
+  return people
+    .filter(person => person.name.toLowerCase().includes(preparedQuery));
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isListShown, setIsListShown] = useState(false);
 
-  return (
-    <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+  const applyDebounceQuery = useCallback(
+    debounce(setDebouncedQuery, 800),
+    [],
+  );
 
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
+  const filteredPeople = useMemo(() => {
+    return search(peopleFromServer, query);
+  }, [debouncedQuery]);
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
+  const onSelect = useCallback(
+    (person: Person) => {
+      setSelectedPerson(person);
+      setIsListShown(false);
+    },
+    [debouncedQuery],
+  );
```

Comment:
task condition :
![image](https://github.com/mate-academy/react_autocomplete/assets/91376823/5a7f341d-183b-4c4a-9189-417ffdabdf99)
in your situation is not saved in input